pilot/pkg/xds/simple.go

```diff
@@ -66,7 +66,7 @@ func NewXDS(stop chan struct{}) *SimpleServer {
 
 	env := model.NewEnvironment()
 	env.Watcher = mesh.NewFixedWatcher(mesh.DefaultMeshConfig())
-	env.PushContext.Mesh = env.Watcher.Mesh()
+	env.PushContext().Mesh = env.Watcher.Mesh()
```

Comment:
this seems wrong, we should not mutate it, instead should create a new PC with Mesh set?

Shouldn't matter in practice though

Reply:
Yes, NewXDS can not be called concurrently.

Reply:
And env is initiated in NewXDS, so no race at all